### DISCUSSION TITLE
How to check whether the decoder is active or not.

### DISCUSSION THREAD

**Discussion by kashish2703** [journeyapps/zxing-android-embedded#598] [Q&A]
#### Description of the problem:

We can check whether the barcode is paused or resumed, but is there a way to check whether the decoder is active or not.
We have a use case where we can start the decoder from multiple entry points, but before starting it, we want to check whether the decoder is already active or not, so that we can avoid starting the decoder again. Is there a way to do the same? Or we need to expose a function.

Thank you!

**Which library version are you using?** E.g. 3.5.0.

**Comment by rkistner** [comment]
There is not currently a public method to check this. That said, BarcodeView does internally keep track of this, and will prevent multiple instances of the decoder running in parallel.

If you need to explicitly keep track of it, it would be best to keep track of this yourself for now.
